packages/cli-repl/src/cli-repl.ts

```diff
@@ -731,7 +739,9 @@ export class CliRepl implements MongoshIOProvider {
     if (analytics) {
       const flushStart = Date.now();
       try {
-        await promisify(analytics.flush.bind(analytics))();
+        await promisify(
+          this.toggleableAnalytics.flush.bind(this.toggleableAnalytics)
+        )();
```

Comment:
Shouldn't this just change `const analytics = this.segmentAnalytics` to const analytics = this.toggleableAnalytics` above? Am I missing something? :)

Reply:
You are not, I completely missed this assignment somehow, thank you!

Reply:
Adjusted in https://github.com/mongodb-js/mongosh/pull/1475/commits/b9263312539c67026c84f5350e9c8d64f83db3c5